bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/FormatCommand.java

```diff
@@ -0,0 +1,111 @@
+
+import static org.apache.bookkeeper.meta.MetadataDrivers.runFunctionWithRegistrationManager;
+
+import com.beust.jcommander.Parameter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.bookkeeper.bookie.Bookie;
+import org.apache.bookkeeper.bookie.BookieException;
+import org.apache.bookkeeper.bookie.Cookie;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.meta.exceptions.MetadataException;
+import org.apache.bookkeeper.tools.cli.helpers.BookieCommand;
+import org.apache.bookkeeper.tools.framework.CliFlags;
+import org.apache.bookkeeper.tools.framework.CliSpec;
+import org.apache.bookkeeper.versioning.Versioned;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+
+public class FormatCommand extends BookieCommand<FormatCommand.Flags> {
+
+    static final Logger LOG = LoggerFactory.getLogger(FormatCommand.class);
+
+    private static final String NAME = "format";
+    private static final String DESC = "Format the current server contents.";
+
+    public FormatCommand() {
+        this(new Flags());
+    }
+
+    public FormatCommand(Flags flags) {
+        super(CliSpec.<Flags>newBuilder()
+            .withName(NAME)
+            .withDescription(DESC)
+            .withFlags(flags)
+            .build());
+    }
+
+    /*
+     * Flags for format bookie command.
+     */
+    @Accessors(fluent = true)
+    @Setter
+    public static class Flags extends CliFlags {
+
+        @Parameter(names = {"-n", "--noninteractive"},
+            description = "Whether to confirm if old data exists?")
+        private boolean nonInteractive;
+
+        @Parameter(names = {"-f", "--force"},
+            description = "If [noninteractive] is specified, then whether"
+                + "to force delete the old data without prompt?")
+        private boolean force;
+
+        @Parameter(names = {"-d", "--deletecookie"},
+            description = "Delete its cookie on metadata store.")
+        private boolean deleteCookie;
+
+    }
+
+    @Override
+    public boolean apply(ServerConfiguration conf, Flags cmdFlags) {
+
+        ServerConfiguration bfconf = new ServerConfiguration(conf);
+        boolean result = Bookie.format(bfconf, cmdFlags.nonInteractive, cmdFlags.force);
+
+        // delete cookie
+        if (cmdFlags.deleteCookie) {
+            try {
+                runFunctionWithRegistrationManager(conf, rm -> {
+
+                    try {
+                        Versioned<Cookie> cookie = Cookie.readFromRegistrationManager(rm, bfconf);
+                        cookie.getValue().deleteFromRegistrationManager(rm, bfconf, cookie.getVersion());
+                    } catch (BookieException.CookieNotFoundException nne) {
+                        LOG.warn("No cookie to remove : ", nne);
+                    } catch (BookieException e) {
+                        e.printStackTrace();
```

Comment:
this will shallow the exception. I would suggest `throw new UncheckedExecutionException(be.getMessage(), be);`